Gems/Multiplayer/Code/Include/Multiplayer/MultiplayerTypes.h

```diff
@@ -189,6 +210,54 @@ namespace Multiplayer
         }
     }
 
+    inline NetworkSpawnable::NetworkSpawnable(const AZ::Data::Asset<AzFramework::Spawnable>& spawnableAsset)
+        : m_spawnableAsset(spawnableAsset)
+    {
+        ;
+    }
+
+    inline void NetworkSpawnable::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<NetworkSpawnable>()
+                ->Version(1)
+                ->Field("Spawnable", &NetworkSpawnable::m_spawnableAsset)
+            ;
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<NetworkSpawnable>("Network Spawnable", "")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "Multiplayer")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &NetworkSpawnable::m_spawnableAsset, "Network Spawnable Asset", "")
+                        ->Attribute(AZ::Edit::Attributes::ChangeValidate, &NetworkSpawnable::ValidatePotentialSpawnableAsset)
+                    ;
+            }
+        }
+    }
+
+    inline AZ::Outcome<void, AZStd::string> NetworkSpawnable::ValidatePotentialSpawnableAsset(void* newValue, const AZ::Uuid& valueType) const
+    {
+        if (azrtti_typeid<AZ::Data::Asset<AzFramework::Spawnable>>() != valueType)
+        {
+            AZ_Assert(false, "Unexpected value type");
+            return AZ::Failure(AZStd::string("Trying to set an network spawnable to something that isn't a spawnable!"));
+        }
+
+        const auto potentialNetworkSpawnable = static_cast<AZ::Data::Asset<AzFramework::Spawnable>*>(newValue);
```

Comment:
This should be a reinterpret_cast in this case since it is casting from a void*

Reply:
I think static_cast is cooler here right; more restrictive = safer? Leave reinterpret_cast for wild-west "convert anything to anything because I know it's weird, but I know what I'm doing". Like when wanting to cast a int* to an float* or something :P 
Thoughts?